packages/core/src/execution-engine/workflow-execute.ts

```diff
@@ -986,6 +989,21 @@ export class WorkflowExecute {
 		return workflowIssues;
 	}
 
+	private getRoutingNodeExecuteFunction(node: INode, type: INodeType) {
+		const { resource, operation } = node.parameters;
+		if (
+			typeof resource === 'string' &&
+			typeof operation === 'string' &&
+			type.nonRoutingOperations?.[resource]?.[operation]
+		) {
+			return type.nonRoutingOperations[resource][operation] as (
+				context: IExecuteFunctions,
+			) => Promise<INodeExecutionData[][] | NodeExecutionWithMetadata[][] | null>;
```

Comment:
Can we dig into why we need to assert the type here? This was already typed [in the workflow package](https://github.com/n8n-io/n8n/pull/12941/files#diff-c5213f7eecd61668679bd4344a9dbcb6a2d737745b159ae3090eea89f0f045a6R1646). 

Also another option:

```ts
private getCustomOperation(node: INode, type: INodeType) {
    const { resource, operation } = node.parameters;

		if (typeof resource !== 'string' || typeof operation !== 'string') return;

    return type.nonRoutingOperations?.[resource]?.[operation];
}
```